hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/ExpressionIndexSupport.scala

```diff
@@ -92,6 +95,47 @@ class ExpressionIndexSupport(spark: SparkSession,
     }
   }
 
+  def prunePartitions(fileIndex: HoodieFileIndex,
```

Comment:
let's call this directly in a test and validate positive and negative scenario

Reply:
Addressed